modules/alloy/src/main/ts/ephox/alloy/behaviour/blocking/BlockingApis.ts

```diff
@@ -0,0 +1,80 @@
+import { Optional, Optionals } from '@ephox/katamari';
+import { Attribute } from '@ephox/sugar';
+import * as BehaviourTypes from '../common/BehaviourTypes';
+import * as Behaviour from '../../api/behaviour/Behaviour';
+import { Keying } from '../../api/behaviour/Keying';
+import { Focusing } from '../../api/behaviour/Focusing';
+import { Replacing } from '../../api/behaviour/Replacing';
+import { AlloySpec } from '../../api/component/SpecTypes';
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import * as GuiFactory from '../../api/component/GuiFactory';
+import { BlockingConfig } from './BlockingSchema';
+import { BlockingState } from './BlockingState';
+
+export interface BlockingApis extends BehaviourTypes.BehaviourApisRecord<BlockingConfig, BlockingState> {
+  block: typeof block;
+  unblock: typeof unblock;
+}
+
+// Mark this component as busy, or blocked.
+const block = (
+  component: AlloyComponent,
+  config: BlockingConfig,
+  state: BlockingState,
+  // This works in conjunction with the 'getRoot' function in the config. To
+  // attach a blocker component to the dom, ensure that 'getRoot' returns a
+  // component, and this function returns the specification of the component to
+  // attach.
+  getBusySpec?: (behaviour: Behaviour.AlloyBehaviourRecord) => AlloySpec
```

Comment:
Why is this optional? This appears to be absolutely required to show the blocker as it has no fallback/default.

- [ ] Make this required unless there's a good reason to have it be optional.

Reply:
I was basing it off of the `getRoot` config field of the `Invalidating` behaviour. I figured that there _might_ be a chance that someone would want to do the visual half of the blocking / unblocking manually, using the callbacks. I know that adding code that we don't currently need is potentially wasteful, but at the same time it would a breaking change (in alloy, not the editor, at least) if we needed to change that later.

Reply:
I've already commented on `getRoot` in the other comments. As for getBusy, having it optional makes it very easy for it to be incorrectly used, so it should be required. Additionally the `on` callbacks are meant to be used to perform other actions, not do the blocking itself.

Reply:
No longer optional